### ISSUE TITLE
No alpha support

### ISSUE THREAD

**Issue by execomrt** [MiYanni/xBRZ.NET#1]
xBRZ should support alpha channel. Currently it is stripped. Is it possible to add it ?

**Comment by MiYanni** [comment]
@execomrt I don't see why not. Can you provide an example image I could use for testing, like those found [here](https://github.com/MiYanni/xBRZ.NET/tree/master/xBRZTester/Images)?

**Comment by execomrt** [comment]
From the xBRZScaler.cs, it says : alpha channel is unused

 -> map source (srcWidth * srcHeight) to target (scale * width x scale * height)
        image, optionally processing a half-open slice of rows [yFirst, yLast) only
        -> color format: ARGB (BGRA char order), alpha channel unused

**Comment by MiYanni** [comment]
@execomrt Correct. I'd have to implement the alpha channel support into the algorithm. Can you provide an image to test with? (Meaning, one that contains an alpha channel that you wish to scale)

**Comment by execomrt** [comment]
![flamealpha](https://user-images.githubusercontent.com/1091621/50162982-824a0e80-02ad-11e9-8116-c72addcd64e2.png)

or https://sourceforge.net/projects/xbrz/files/latest/download (there are a lot of images).

I've tried to changed on my side, but since you used signed int for bitmap element, there were a lot of refactor. Check the .CPP code, there is a special case for handling distance of alpha value (source code is on Sourceforge.net, updated in Mars 2018)(

**Comment by MiYanni** [comment]
@execomrt If you want to create a pull request, I can at least take a look at the changes you were doing. Maybe tweak them and see if I can reduce the refactoring required.
